code/_core/obj/structure/totem.dm

```diff
@@ -31,7 +33,8 @@
 	if(world.time <= next_fire)
 		return TRUE
 	next_fire = world.time + cooldown_fire
-	totemic_effect()
+	if(owner in viewers(11,loc))
```

Comment:
in viewers is extremely expensive, even if it's called every second.

Instead you should use ``get_dist(owner,src) <= VIEW_RANGE``